### PR TITLE
Add Quant Analyzer & YoLov6 Demo

### PR DESCRIPTION
**Pull request by RachelXu7**
Quant Analyzer provides:
1. compute the sensitivity of the model by quantizing layer by layer; (sensitivity is expressed as model accuracy for now)
2. rank the sensitivity which can show which layer is not suitable for quantization
3. plot the histogram of tensors; (you can choose the number of tensors you want to visualize)
4. automatically load and save checkpoint since there may be too many layers to be analyzed and it will be sad if the program breaks off

### CODE REVIEW

File: paddleslim/quant/utils.py [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys
+import numpy as np
+from paddle.fluid.framework import IrNode
+from paddle.fluid.framework import Operator
+
+_op_real_in_out_name = {
```
Comment (yghstill) [inline paddleslim/quant/utils.py]: 复用paddle.fluid.contrib.slim.quantization.utils里的即可

File: paddleslim/quant/utils.py [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+        for i, data in enumerate(dataloader()):
+            if not isinstance(data, Iterable):
+                data = [data]
+            yield dict((name_, np.array(data_))
+                       for name_, data_ in zip(names, data))
+
+    return gen
+
+
+def _get_op_input_var_names(op):
```
Comment (yghstill) [inline paddleslim/quant/utils.py]: same as above

File: paddleslim/quant/analysis.py [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import sys
+import pickle
+import copy
+import logging
+import matplotlib.pyplot as plt
```
Comment (yghstill) [inline paddleslim/quant/analysis.py:20]: add in requirements.txt

**Review by yghstill — APPROVED** [review]
后续TODO：
- 增加文档说明
- 增加单测

**Review by ceci3 — COMMENTED** [review]
这个示例直接放在example 里之前的pytorch_yolov6里面？量化和量化分析之间有一个顺序关系，个人感觉在一个目录会比较好

File: paddleslim/quant/analysis.py [hunk 4]
```diff
@@ -0,0 +1,358 @@
...
+import numpy as np
+
+import paddle
+from paddle.fluid import core
+from paddle.fluid import framework
+from paddle.fluid.framework import IrGraph
+from paddle.fluid.executor import Executor, global_scope
+from paddle.fluid.contrib.slim.quantization import PostTrainingQuantization
+from paddleslim.quant.quanter import quant_post
+from paddleslim.core import GraphWrapper
```
Comment (ceci3) [inline paddleslim/quant/analysis.py:31]: 改成相对路径

File: paddleslim/quant/analysis.py [hunk 5]
```diff
@@ -0,0 +1,358 @@
...
+from paddle.fluid.contrib.slim.quantization.utils import _get_op_input_var_names, load_variable_data
+from paddleslim.common import get_logger
```
Comment (ceci3) [inline paddleslim/quant/analysis.py:33]: 同上

File: paddleslim/quant/analysis.py [hunk 6]
```diff
@@ -0,0 +1,358 @@
...
+
+        if not os.path.exists(self.save_dir):
+            os.mkdir(self.save_dir)
+
+        devices = paddle.device.get_device().split(':')[0]
+        self.places = paddle.device._convert_to_place(devices)
+        executor = paddle.static.Executor(self.places)
+
+        # load model 
+        [program, self.feed_list, self.fetch_list]= paddle.fluid.io.load_inference_model( \
```
Comment (ceci3) [inline paddleslim/quant/analysis.py:132]: fluid -> static

File: paddleslim/quant/analysis.py [hunk 7]
```diff
@@ -0,0 +1,358 @@
...
+import logging
+import matplotlib.pyplot as plt
+from matplotlib.backends.backend_pdf import PdfPages
+import numpy as np
+
+import paddle
+from paddle.fluid import core
+from paddle.fluid import framework
+from paddle.fluid.framework import IrGraph
+from paddle.fluid.executor import Executor, global_scope
```
Comment (ceci3) [inline paddleslim/quant/analysis.py:28]: 这个应该可以直接从static里import，不用从fluid里import

File: paddleslim/quant/analysis.py [hunk 8]
```diff
@@ -0,0 +1,358 @@
...
+
+    def calculate_histogram(self):
+        '''
+        Sample histograms for the weight and corresponding act tensors
+        '''
+        devices = paddle.device.get_device().split(':')[0]
+        places = paddle.device._convert_to_place(devices)
+        executor = paddle.static.Executor(places)
+
+        [program, feed_list, fetch_list]= paddle.fluid.io.load_inference_model( \
```
Comment (ceci3) [inline paddleslim/quant/analysis.py:279]: fluid -> static

**Pull request closed without merging**